File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& v, int k) {
        int n=v.size();
        int s=0;
        for(int i=0;i<n;i++){
            s+=v[i];
            if(s>k) return i;
        }
        k=k%s;
        for(int i=0;i<n;i++){
            if(v[i]>k) return i;
            k-=v[i];
        }
        return -1;
    }
};

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] c, int k) {
        int n=c.length;
        int s=0;
        for(int i=0;i<n;i++){
            s+=c[i];
            if(s>k) return i;
        }
        k=k%s;
        for(int i=0;i<n;i++){
            if(k<c[i]) return i;
            k-=c[i];
        }
        return -12;
    }
}